src/Persistent/Pure.hs

```diff
@@ -606,6 +608,31 @@ addDelegation = addDb dbDelegationMap
 deleteDelegation :: AUID Delegation -> AUpdate ()
 deleteDelegation did = dbDelegationMap . at did .= Nothing
 
+delegationScopeTree :: User -> Query (Tree DScopeFull)
+delegationScopeTree user = unfoldTreeM discover DScopeGlobalFull
+  where
+    discover :: DScopeFull -> Query (DScopeFull, [DScopeFull])
+    discover DScopeGlobalFull = do
+        schoolIdeas <- DScopeIdeaFull <$$> findWildIdeasBySpace SchoolSpace
+        let ideaSpaces = DScopeIdeaSpaceFull . ClassSpace <$> (user ^.. userSchoolClasses)
+        pure (DScopeGlobalFull, schoolIdeas <> ideaSpaces)
+
+    -- FIXME: Impossible: As the Global is the School space
+    discover s@(DScopeIdeaSpaceFull SchoolSpace) = do
```

Comment:
this should be removed, and the next case should cover this (just relax the pattern).